### PR TITLE
[CALCITE] Added support for CREATE TABLE attribute MERGEBLOCKRATIO

### PR DESCRIPTION
**Pull request by michaelvdow**

### CODE REVIEW

File: babel/src/main/codegen/includes/parserImpls.ftl [hunk 1]
```diff
@@ -236,6 +236,31 @@ SqlCreateAttribute CreateTableAttributeJournalTable() :
...
+SqlCreateAttribute CreateTableAttributeMergeBlockRatio() :
+{
+    MergeBlockRatioModifier modifier = MergeBlockRatioModifier.UNSPECIFIED;
+    int ratio = 1;
+    boolean percent = false;
+}
+{
+    (
+        (
+            <DEFAULT_> { modifier = MergeBlockRatioModifier.DEFAULT; }
```
Comment (AndrewPochapsky) [inline babel/src/main/codegen/includes/parserImpls.ftl:248]: Is the underscore in the token supposed to be there?

Comment (michaelvdow) [reply babel/src/main/codegen/includes/parserImpls.ftl:248]: Yeah <DEFAULT_> maps to the "DEFAULT" token. I think just <DEFAULT> maps to something  else.

File: babel/src/main/codegen/includes/parserImpls.ftl [hunk 2]
```diff
@@ -236,6 +236,31 @@ SqlCreateAttribute CreateTableAttributeJournalTable() :
...
+        |
+            <NO> { modifier = MergeBlockRatioModifier.NO; }
+        )
+        <MERGEBLOCKRATIO>
+    |
+        <MERGEBLOCKRATIO> <EQ> ratio = UnsignedIntLiteral()
+        [ <PERCENT> { percent = true; } ]
+    )
+    {
+        if (ratio >= 1 && ratio <= 100) {
```
Comment (dasch-1) [inline babel/src/main/codegen/includes/parserImpls.ftl:258]: Suggest making this an explicit parser failure by throwing an exception, such as is done here:
https://github.com/googleinterns/calcite/blob/master/core/src/main/codegen/templates/Parser.jj#L404

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java [hunk 3]
```diff
@@ -356,6 +356,36 @@ private void checkParseInfixCast(String sqlType) {
...
+    final String sql = "create table foo, mergeblockratio = ^-^20 (bar integer)";
+    final String expected = "(?s).*Encountered \"-\" at .*";
+    sql(sql).fails(expected);
+  }
+
+  @Test public void testTableAttributeMergeBlockRatioPercent() {
+    final String sql = "create table foo, mergeblockratio = 45 percent (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, MERGEBLOCKRATIO = 45 PERCENT (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
```
Comment (dasch-1) [inline babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:387]: Should we have a test case where the ratio is outside of the bounds of 1 to 100 (but not negative)?

Comment (michaelvdow) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:387]: Sure I can add that in

File: babel/src/main/codegen/includes/parserImpls.ftl (hunk 2 above)
Comment (michaelvdow) [reply babel/src/main/codegen/includes/parserImpls.ftl:258]: Sure, I added an exception thrown for number literal out of range

File: babel/src/test/java/org/apache/calcite/test/BabelParserTest.java [hunk 4]
```diff
@@ -356,6 +356,36 @@ private void checkParseInfixCast(String sqlType) {
...
+
```
Comment (dstekol) [inline babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:394]: Maybe also worth adding a test to show that if you use a modifier keyword before MERGEBLOCKRATIO and an equals assignment after it, then parsing fails (like "NO MERGEBLOCKRATIO = 10" ) ?

Comment (michaelvdow) [reply babel/src/test/java/org/apache/calcite/test/BabelParserTest.java:394]: That's a good point, I can add another test for that

**Pull request merged**